tcms/rpc/tests/test_testplan.py

```diff
@@ -226,3 +227,116 @@ def test_ignores_existing_mappings(self):
+
+    def test_create_plan_with_empty_required_field(self):
+        self.params['name'] = ''
+        with self.assertRaisesRegex(XmlRPCFault, 'This field is required'):
+            self.rpc_client.TestPlan.create(self.params)
+
+    def test_create_plan_without_permissions(self):
+        unauthorized_user = UserFactory()
+        unauthorized_user.set_password('api-testing')
+        unauthorized_user.save()
+
+        unauthorized_user.user_permissions.add(*Permission.objects.all())
+        remove_perm_from_user(unauthorized_user, 'testplans.add_testplan')
+
+        rpc_client = xmlrpc.TCMSXmlrpc(unauthorized_user.username,
+                                       'api-testing',
+                                       '%s/xml-rpc/' % self.live_server_url).server
+
+        with self.assertRaisesRegex(ProtocolError, '403 Forbidden'):
+            rpc_client.TestPlan.create(self.params)
+
+
+class TestListAttachments(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.product = ProductFactory()
+        self.version = VersionFactory(product=self.product)
+        self.plan = TestPlanFactory(product_version=self.version,
+                                    product=self.product,
+                                    author=UserFactory(),
+                                    type=PlanTypeFactory(name='manual smoking'))
+
+    def test_list_attachments(self):
+        attachments = self.rpc_client.TestPlan.list_attachments(self.plan.pk)
+        self.assertIsNotNone(attachments)
+        self.assertIsInstance(attachments, list)
+
+    def test_list_attachments_with_wrong_plan_id(self):
+        with self.assertRaisesRegex(XmlRPCFault, 'TestPlan matching query does not exist'):
+            self.rpc_client.TestPlan.list_attachments(-1)
+
+    def test_list_attachments_without_permissions(self):
+        unauthorized_user = UserFactory()
+        unauthorized_user.set_password('api-testing')
+        unauthorized_user.save()
+
+        unauthorized_user.user_permissions.add(*Permission.objects.all())
+        remove_perm_from_user(unauthorized_user, 'attachments.view_attachment')
+
+        rpc_client = xmlrpc.TCMSXmlrpc(unauthorized_user.username,
+                                       'api-testing',
+                                       '%s/xml-rpc/' % self.live_server_url).server
+
+        with self.assertRaisesRegexp(ProtocolError, '403 Forbidden'):
+            rpc_client.TestPlan.list_attachments(self.plan.pk)
+
+
+class TestAddAttachment(APITestCase):
+    def _fixture_setup(self):
+        super()._fixture_setup()
+
+        self.product = ProductFactory()
+        self.version = VersionFactory(product=self.product)
+        self.plan = TestPlanFactory(product_version=self.version,
+                                    product=self.product,
+                                    author=UserFactory(),
+                                    type=PlanTypeFactory(name='manual smoking'))
```

Comment:
You don't need the other fields/factories here. TestPlanFactory will create everything that it needs when not specified.